department-of-veterans-affairs/caseflow
# MattT/APPEALS-9100: Change BVA Intake's "Return appeal to VHA" to "Return appeal to VHA CAMO"

Resolves [APPEALS-9100](https://vajira.max.gov/browse/APPEALS-9100)

### Description
With the addition of the VHA Caregiver Support Program and its associated pre-docket workflow, we would like to be more specific in regards to signaling which organization BVA Intake is interacting with since VHA CAMO and VHA Caregiver Support Program are both a part of "VHA".

### Acceptance Criteria
Whenever an appeal that has been to VHA CAMO is returned to BVA Intake (either as ready for review or due to an issue with the appeal)

- [ ] There is a task action in BVA Intake's actions dropdown that reads "Return appeal to VHA CAMO". Currently it reads "Return appeal to VHA".
 - [ ] Upon selecting the task action from the dropdown:
   - [ ] A modal appears
      - [ ] The modal title: "Return to VHA CAMO"
      - [ ] The modal description: "If you are unable to docket this appeal due to insufficient documentation, you may return this to VHA CAMO."
 - [ ] Upon enacting the task action:
   - [ ] The appeal will be located in BVA Intake's "Pending" tab, and also in VHA CAMO's "Assigned" tab".
   - [ ] The BVA Intake user will be redirected back to their "Pending" tab.
   - [ ]  The BVA Intake user will be presented with a success banner that reads: "You have successfully returned <appellant's name>'s case to VHA CAMO"

### Testing Plan
1. Open the Rails console, and stage an appeal that has a VhaDocumentSearchTask that has been to VHA CAMO and sent back to BVA Intake:
```ruby
appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
parent_task = appeal.tasks.last
camo_task = VhaDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: VhaCamo.singleton )
camo_task.completed!
```
2. Log into Caseflow as an Intake user (ex: `BVADWISE`)
3. Navigate to the Case Details page for the appeal. You can use the Rails console to obtain the appeal's UUID for the URL path.
```ruby
appeal.uuid
```
Path: `/queue/appeals/<appeal.uuid>`
4. Make sure that the following option is in the actions dropdown: "Return to VHA CAMO". Click on that option.
5. Whenever the modal appears, make sure that:
> The title is: "Return to VHA CAMO"
> The description text is: "If you are unable to docket this appeal due to insufficient documentation, you may return this to VHA CAMO."
6. Enter text into the text field and then click the "Submit" button.
7. You should be redirected back to BVA Intake's Pending tab.
8. A green success banner should appear on the page, and say: "You have successfully returned <appellant's name>'s case to VHA CAMO"


### User Facing Changes

 ITEM|BEFORE|AFTER
 ---|---|---
 Actions Dropdown|![](https://i.imgur.com/aM5XszD.png)|<img width="771" alt="bva-to-camo-dropdown-post-fix" src="https://user-images.githubusercontent.com/99351305/190032182-28af652b-023f-4779-9167-53641cd198e2.PNG">
 Modal|![](https://i.imgur.com/DGwsHIs.png)|<img width="336" alt="bva-to-camo-modal" src="https://user-images.githubusercontent.com/99351305/190032204-930b7c7f-365f-40d3-817a-a5237761abfe.PNG">
 Success Banner|![](https://i.imgur.com/TK7sJVf.png)|<img width="812" alt="bva-to-camo-success-banner-post-fix" src="https://user-images.githubusercontent.com/99351305/190032231-f93b7a9e-0d12-4868-b45f-21199e679ee3.PNG">

### Code Documentation Updates
